package proto_ff;

// calm 自定义的fieldoption都在该文件中
import "yd_fieldoptions.proto";
import "Com.proto";
import "ComDefine.proto";
import "DBProto.proto";


// 角色DB数据
message GetRegisterNum_RoleDBData
{
  optional uint32 num = 1;
}

message RoleDBSimpleData
{
  optional uint64                 cid					= 1;	// role cid
  optional uint32                 zid					= 2;    // 区服ID
  optional uint32					        uid					= 3;    // UID
  optional RoleDBBaseData         base				= 4;    // 基础数据
}

message MyFriendInfo
{

}

message RoleDBSnsDetail
{
  optional uint64               cid	= 1[(yd_fieldoptions.db_field_type) = E_FIELDTYPE_PRIMARYKEY];	// role cid
  optional MyFriendInfo         friend_info = 2;
}

message RoleEnterSceneData
{
  optional uint64                 cid					= 1;	// role cid
  optional uint32                 zid					= 2;    // 区服ID
  optional uint32					        uid					= 3;    // UID
  optional RoleDBBaseData         base				= 10;    // 基础数据
  optional AttrDBData             attr        = 11;
}